### PR TITLE
add docs for listing tokens

### PR DESCRIPTION
**Pull request by tomsik68**
As per Colton's request, I'm adding some documentation for listing tokens for an application or identity. Let me know if I need to elaborate somewhere or reword things.

### CODE REVIEW

**Comment by tomsik68** [comment]
cc @beyondchojnacki @turnipdabeets can you guys take a look as well?

File: docs/workflows/api-token.md [hunk 1]
```diff
@@ -546,3 +546,48 @@ In order to validate a token offline, the JWT header and claims must be decoded.
...
+
+You need to use a Management API Bearer token with the `tokens:read` scope in order to call this API.
+
+### Request: Listing access tokens for an identity
+
+```bash
+curl 'https://api-${REGION}.beyondidentity.run/v1/tenants/${TENANT_ID}/realms/${REALM_ID}/tokens?application=${APPLICATION_ID}&principal_type=identity&principal_id=${IDENTITY_ID}'
+-H "Authorization Bearer ${MANAGEMENT_API_TOKEN}"
+```
+
```
Comment (beyondchojnacki) [inline docs/workflows/api-token.md:563]: Can we add an example response here? I know they may look the same, but the doc is a bit confusing if we dont explicitly show them the response

File: docs/workflows/api-token.md [hunk 2]
```diff
@@ -546,3 +546,48 @@ In order to validate a token offline, the JWT header and claims must be decoded.
 4. Check timestamps in JWT claims where `nbf` <= current time as unix timestamp in seconds <= `exp`
 5. Check that JWT claims target tenant `bi_t` and target realm `bi_r` match the tenant and realm for the given application.
 6. Check that JWT claims has the expected scopes.
+
+## Listing access tokens
```
Comment (turnipdabeets) [inline docs/workflows/api-token.md:550]: For consistency with the other titles 
```suggestion
## Listing Access Tokens
```

File: docs/workflows/api-token.md [hunk 3]
```diff
@@ -546,3 +546,48 @@ In order to validate a token offline, the JWT header and claims must be decoded.
...
+
+### Request: Listing access tokens for an application
+
+```bash
+curl 'https://api-${REGION}.beyondidentity.run/v1/tenants/${TENANT_ID}/realms/${REALM_ID}/tokens?application=${APPLICATION_ID}&principal_type=application&principal_id=${APPLICATION_ID}'
+-H "Authorization Bearer ${MANAGEMENT_API_TOKEN}"
+```
+
+In this case, the application is both the issuer and the subject of the token,
+which is why it needs to be in the request twice.
```
Comment (turnipdabeets) [inline docs/workflows/api-token.md:572]: I would move this up to under the header & above the bash example 
I would also add a similar comment for the "Listing access tokens for an identity" as the `principal_id`

```suggestion
When listing access tokens for an application, the application need to be in the request twice as it is both the issuer `application` and the subject `principal_id` of the token. 
```

File: docs/workflows/api-token.md [hunk 4]
```diff
@@ -546,3 +546,48 @@ In order to validate a token offline, the JWT header and claims must be decoded.
...
+
+### Example response
```
Comment (turnipdabeets) [inline docs/workflows/api-token.md:574]: If this is the same example response for both, you could add a comment here that the response for listing access tokens for either application or identity will return the same response.

File: docs/workflows/api-token.md [hunk 5]
```diff
@@ -546,3 +546,48 @@ In order to validate a token offline, the JWT header and claims must be decoded.
...
+## Listing access tokens
+
+Our API also provides the functionality to list active tokens that are held by a given identity or an application.
+In order to list tokens, you need to know the ID of the application that was used to issue the token.
+
+You need to use a Management API Bearer token with the `tokens:read` scope in order to call this API.
+
+### Request: Listing access tokens for an identity
+
+```bash
```
Comment (turnipdabeets) [inline docs/workflows/api-token.md:559]: I was using https://curlconverter.com to create the other language code snippets from curl and then wrapping them in the <Tabs>. I can help with this part if you want or need.

**Review by turnipdabeets — CHANGES_REQUESTED** [review]

**Comment by turnipdabeets** [comment]
> cc @beyondchojnacki @turnipdabeets can you guys take a look as well?

Thanks for doing this!

File: docs/workflows/api-token.md (hunk 4 above)
Comment (turnipdabeets) [inline docs/workflows/api-token.md:574]:
```suggestion
### Example Response
```

File: docs/workflows/api-token.md (hunk 5 above)
Comment (turnipdabeets) [reply docs/workflows/api-token.md:559]: I just updated this file with a `MultiLanguageCodeBlock` to handles this. If you merge main you should see how to add the curl to this block.

Comment (tomsik68) [reply docs/workflows/api-token.md:559]: Thank you so much, this is excellent!

File: docs/workflows/api-token.md (hunk 4 above)
Comment (tomsik68) [reply docs/workflows/api-token.md]: I added the response to both sections. I think that's least confusing as the responses are also a little bit different.

File: docs/workflows/api-token.md [hunk 6]
```diff
@@ -135,3 +135,75 @@ In order to validate a token offline, the JWT header and claims must be decoded.
...
+### Request: Listing access tokens for an identity
+
+In this case, the application is the issuer of the token and the identity is the subject of the token.
+
+<MultiLanguageCodeBlock
+curl='
+curl "https://api-${REGION}.beyondidentity.run/v1/tenants/${TENANT_ID}/realms/${REALM_ID}/tokens?application=${APPLICATION_ID}&principal_type=identity&principal_id=${IDENTITY_ID}"
+-H "Authorization Bearer ${MANAGEMENT_API_TOKEN}"
+' title="/tokens"
+/>
```
Comment (turnipdabeets) [inline docs/workflows/api-token.md:155]:
```suggestion
<MultiLanguageCodeBlock
curl='curl "https://api-$(REGION).beyondidentity.run/v1/tenants/$(TENANT_ID)/realms/$(REALM_ID)/tokens?application=$(APPLICATION_ID)&principal_type=identity&principal_id=$(IDENTITY_ID)" \
-H "Authorization Bearer $(MANAGEMENT_API_TOKEN)"' 
title="/tokens"
/>
```

File: docs/workflows/api-token.md [hunk 7]
```diff
@@ -135,3 +135,75 @@ In order to validate a token offline, the JWT header and claims must be decoded.
...
+
+In this case, the application is both the issuer and the subject of the token,
+which is why it needs to be in the request twice.
+
+<MultiLanguageCodeBlock
+curl='
+curl "https://api-${REGION}.beyondidentity.run/v1/tenants/${TENANT_ID}/realms/${REALM_ID}/tokens?application=${APPLICATION_ID}&principal_type=application&principal_id=${APPLICATION_ID}"
+-H "Authorization Bearer ${MANAGEMENT_API_TOKEN}"
+' title="/tokens"
+/>
```
Comment (turnipdabeets) [inline docs/workflows/api-token.md:188]: And here too
```suggestion
<MultiLanguageCodeBlock
curl='curl "https://api-$(REGION).beyondidentity.run/v1/tenants/$(TENANT_ID)/realms/$(REALM_ID)/tokens?application=$(APPLICATION_ID)&principal_type=application&principal_id=$(APPLICATION_ID)" \
-H "Authorization Bearer $(MANAGEMENT_API_TOKEN)"' 
title="/tokens"
/>
```

**Review by turnipdabeets — CHANGES_REQUESTED** [review]
Looks good to me! Just a few changes in the code block to make it look like the rest of the document.

File: docs/workflows/api-token.md [hunk 8]
```diff
@@ -135,3 +135,75 @@ In order to validate a token offline, the JWT header and claims must be decoded.
...
+      "scope": "",
+      "exp": 1675177245,
+      "iat": 1675090845,
+      "token_type": "access",
+      "principal_path": "tenants/0001fb357ac698bb/realms/d7e6e18f570421ce/applications/43e65451-1a38-4e17-8fd3-ae5b582fd9c7",
+      "token_format": "self_contained"
+    }
+  ]
+}
+
```
Comment (turnipdabeets) [inline docs/workflows/api-token.md:207]: Removing extra space
```suggestion
```

File: docs/workflows/api-token.md [hunk 9]
```diff
@@ -135,3 +135,75 @@ In order to validate a token offline, the JWT header and claims must be decoded.
...
+      "scope": "",
+      "exp": 1675177245,
+      "iat": 1675090845,
+      "token_type": "access",
+      "principal_path": "tenants/0001fb357ac698bb/realms/d7e6e18f570421ce/identities/ce9d40cf7285a260",
+      "token_format": "self_contained"
+    }
+  ]
+}
+
```
Comment (turnipdabeets) [inline docs/workflows/api-token.md:174]: Removing extra space
```suggestion
```

**Comment by tomsik68** [comment]
Thanks for taking a look, Anna! Let's wait for Colton's review. I'm curious what he thinks.

**Review by turnipdabeets — APPROVED** [review]

File: docs/workflows/api-token.md (hunk 1 above)
Comment (tomsik68) [reply docs/workflows/api-token.md:155]: I added an example response to both cases. Would it make sense like this?

**Comment by beyondchojnacki** [comment]
Looking at the token revocation API. Is the developer supposed to use the token id from list tokens to revoke? Can we make that more clear?

I am thinking through a developer trying to implement the following in an account settings page for their application:
1) As an end user I would like to see a list of my active sessions in my account settings page
2) As an end user I would like to select an active session and revoke it (from my account settings page)

**Comment by beyondchojnacki** [comment]
Looks good to me

**Pull request merged**